### PR TITLE
Upgrade Core & Common

### PR DESCRIPTION
**Pull request by macdrevx**
Fixes: #571

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Update the changelog

### Summary of changes

- Updates to MapboxCoreMaps v10.0.0 and MapboxCommon v20.0.0.

#### Public API Changes

- Core and Common APIs that accept user-defined implementations of protocols now hold strong references to the provided objects. Please audit your usage of the following protocols and make any required changes to avoid memory leaks: `CustomLayerHost`, `ElevationData`, `MapClient`, `MBMMetalViewProvider`, `Observer`, `OfflineRegionObserver`, `HttpServiceInterceptorInterface`, `HttpServiceInterface`, `LogWriterBackend`, `OfflineSwitchObserver`, `ReachabilityInterface`, `TileStoreObserver`

#### Other changes

- Updates `CustomLayerExample`, `OfflineRegionManagerExample`, and `SceneKitExample` to avoid memory leaks
- `MapView`'s stored reference to `mapClient` is no longer necessary and has been removed.
- `eventHandlers` properties of `MapboxMap` and `Snapshotter` are now `let`s.
- Reorganized some comments
- Removed some unnecessary MARK separating lines
- Disabled part of a `MigrationGuideIntegrationTests` test case that invoked `HttpServiceFactory.setUserDefinedForCustom` to avoid modifying shared global state that could impact other tests.
- Deleted unused `DelegatingObserver` and associated delegate protocol

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +38,7 @@ Mapbox welcomes participation and contributions from everyone.
 * `BasicCameraAnimator` no longer updates the camera a final time after being stopped or canceled prior to running to completion. ([#712](https://github.com/mapbox/mapbox-maps-ios/pull/712))
 * `BasicCameraAnimator.isReversed` is now settable. ([#712](https://github.com/mapbox/mapbox-maps-ios/pull/712))
 * The double tap, quick zoom, and double touch gestures now use the gesture's location in the view to anchor camera changes. Previously, they used the camera's center coordinate ([#722](https://github.com/mapbox/mapbox-maps-ios/pull/722))
+* `MapboxCommon.HTTPServiceFactor.reset()` has been added to release the HTTP service implementation ([#732](https://github.com/mapbox/mapbox-maps-ios/pull/732))
```
Comment (neelmistry94) [inline CHANGELOG.md:41]: Missing period before pr link

File: CHANGELOG.md [hunk 2]
```diff
@@ -23,6 +23,9 @@ Mapbox welcomes participation and contributions from everyone.
...
+* `MapboxCoreMaps.Settings` is now deprecated. ([#732](https://github.com/mapbox/mapbox-maps-ios/pull/732))
+* Setting `data` property on a GeoJSON source via `Style.setSourceProperty(for:property:value:)` or `Style.updateGeoJSONSource(withId:geoJSON:)` is now asynchronous and never returns an error. Errors will be reported asynchronously via a `MapEvents.EventKind.mapLoadingError` event instead. ([#732](https://github.com/mapbox/mapbox-maps-ios/pull/732))
+* Core and Common APIs that accept user-defined implementations of protocols now hold strong references to the provided objects. Please audit your usage of the following protocols and make any required changes to avoid memory leaks: `CustomLayerHost`, `ElevationData`, `MapClient`, `MBMMetalViewProvider`, `Observer`, `OfflineRegionObserver`, `HttpServiceInterceptorInterface`, `HttpServiceInterface`, `LogWriterBackend`, `OfflineSwitchObserver`, `ReachabilityInterface`, `TileStoreObserver` ([#732](https://github.com/mapbox/mapbox-maps-ios/pull/732))
```
Comment (neelmistry94) [inline CHANGELOG.md:28]: Missing period before pr link

File: Sources/MapboxMaps/Snapshot/Snapshotter.swift [hunk 3]
```diff
@@ -253,10 +254,35 @@ public class Snapshotter {
...
 
+    /// Unsubscribes an Observer from a provided list of event types.
+    ///
+    /// A type conforming to `ObservableProtocol` will hold a strong reference
+    /// to an Observer instance, therefore, in order to stop receiving notifications,
+    /// the caller must call `unsubscribe` with the instance used for an initial
+    /// subscription.
+    ///
+    /// - Parameters:
+    ///   - observer: An `Observer`
```
Comment (neelmistry94) [inline Sources/MapboxMaps/Snapshot/Snapshotter.swift:283]: This doesn't feel like a very useful comment. What is it observing?

File: Tests/MapboxMapsTests/MigrationGuide/MigrationGuideIntegrationTests.swift [hunk 4]
```diff
@@ -202,7 +202,11 @@ class MigrationGuideIntegrationTests: IntegrationTestCase {
             let customHTTPService = CustomHttpService()
 
             func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-                HttpServiceFactory.setUserDefinedForCustom(customHTTPService)
+                // commenting this out for testing purposes to avoid
+                // interfering with global state that may impact other
+                // tests
+                //
+                // HttpServiceFactory.setUserDefinedForCustom(customHTTPService)
```
Comment (neelmistry94) [inline Tests/MapboxMapsTests/MigrationGuide/MigrationGuideIntegrationTests.swift:209]: Is this meant to merge or was this for some experimentation

**Review by neelmistry94 — COMMENTED** [review]
Some comments and feedback

File: Tests/MapboxMapsTests/MigrationGuide/MigrationGuideIntegrationTests.swift (hunk 4 above)
Comment (macdrevx) [reply Tests/MapboxMapsTests/MigrationGuide/MigrationGuideIntegrationTests.swift:209]: Meant to merge. I will open a ticket to follow up in a better way later.

File: Sources/MapboxMaps/Snapshot/Snapshotter.swift (hunk 3 above)
Comment (macdrevx) [reply Sources/MapboxMaps/Snapshot/Snapshotter.swift:283]: I'd like to ticket this. I merely relocated the existing documentation.

Comment (macdrevx) [reply Sources/MapboxMaps/Snapshot/Snapshotter.swift]: Nevermind, I made some updates to these comments

**Review by neelmistry94 — APPROVED** [review]

**Pull request merged**